Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -732,6 +732,14 @@ class Gx2Fitter {
     BoundMatrix aMatrix = BoundMatrix::Zero();
     BoundVector bVector = BoundVector::Zero();
 
+    // We need to create a temporary track container. We create several times a
+    // new track and delete it after updating the parameters. However, if we
+    // would work on the externally provided track container, it would be
+    // difficult to remove the correct track, if it contains more than one.
+    track_container_t tc;
+    traj_t traj;
+    TrackContainer trackContainerTemp{tc, traj};
```

Comment:
This then mirrors the output backend, which can potentially be negative for performance. Should we hard-code the transient backends here?